### PR TITLE
Turk-on-call smarter bug report

### PR DESCRIPTION
**Pull request by mialsy**
# Description

Smarter bug report for Turk-on-call(TAO) jobs. Added JobListener and DataGenerator for processing log files for TAO job and upload the processed file to s3.

- The class TaoLogListener(JobListener) listens to the status file for a TAO job batch, and initiate a bug report job once logs are ready.
- The class TaoBugReportJob(DataGenerator) read in the logs for a TAO job batch, extract traceback, and output the content of the traceback and the frequency of the traceback to s3. 

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] Proposes a change (non-breaking change that isn't necessarily a bug)
- [ ] Refactor
- [x] New feature (non-breaking change that adds a new functionality)
- [ ] Breaking change (fix or feature that would break some existing functionality downstream)
- [ ] This is a unit test
- [ ] Documentation only change
- [ ] Datasets Release
- [ ] Models Release

## Type of requested review

- [x] I want a thorough review of the implementation.
- [ ] I want a high level review. 
- [ ] I want a deep design review.

## Before and After

***Before***:
When a turk-on-call job batch finished, it would have one compressed log.tar.gz file. Only raw log file would be included in the compressed file.

***After***:
Besides the log.tar.gz file, a log_traceback.csv would be stored on s3, which include the bug traceback detail in the logs and the count of the same bug appearance in this batch. 

After running with a batch id passed in, will see the following:
- The traceback file located on s3:
  Screen Shot 2022-06-10 at 2 20 21 PM
- The traceback csv file content:
  Screen Shot 2022-06-10 at 2 19 51 PM

# Testing

Tested with manually copied a stat file to s3, and then run the `oncall_bug_report.py` passing in a batch id. 

# Checklist:

- [x] I have performed manual end-to-end testing of the feature in my environment.
- [x] I have added Docstrings and comments to the code.
- [ ] I have made changes to existing documentation where needed.
- [ ] I have added tests that show that the PR is functional.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have added relevant collaborators to review the PR before merge.
- [ ] [Polymetis only] I ran on hardware (1) all scripts in `tests/scripts`, (2) asv benchmarks.

### CODE REVIEW

**Review by snyxan — COMMENTED** [review]
Thanks for creating this PR! Could you please add a short README to explain the high level idea of the module you added?

File: droidlet/tools/hitl/turk_oncall/oncall_bug_report.py [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+        "--tao_job_batch_id",
+        type=int,
+        required=True,
+        help="TAO job batch id",
+    )
+    opts = parser.parse_args()
+
+    runner = TaskRunner()
+
+    class MockDataGenerator(DataGenerator):
```
[the remark this answers is not in the document]
Comment (mialsy) [reply droidlet/tools/hitl/turk_oncall/oncall_bug_report.py:259]: Make sense! I will move this mock piece to a test folder.

**Comment by mialsy** [comment]
> Thanks for creating this PR! Could you please add a short README to explain the high level idea of the module you added?

Yup! I just added a README (`droidlet/tools/hitl/turk_oncall/README.MD`), let me know if this looks okay.

**Review by snyxan — APPROVED** [review]

**Comment by snyxan** [comment]
LGTM! Ship it once the conflicts are resolved.

**Pull request merged**